### ISSUE TITLE
-ios-signing-cert

### ISSUE THREAD

**Issue by SwiftyFlow** [flutter/flutter#97964]
2 days wasted at trying to fix and find why flutter is picking up the wrong signing certificate. I have tried everything I could find on this not documented bug.

flutter config --clear-ios-signing-cert does clear anything

flutter channel master
flutter upgrade
flutter clean
flutter build ios

Doesn't do much, we get beautiful lines of terminal but after that not much.

Done my things directly in Xcode which by the way showed exactly the signing id it needed (back then). But Flutter knows better what to use to fail.

I have done multiple more trials at random things. Nothing works...

I even deleted all my certificates via 'security find-identity -p codesigning -v' and re-added only the one I wanted it still downloads a wrong one and set the project to it.

Still seaarching where is this file vim ~/.flutter_settings
To finally put:
{
  "ios-signing-cert": "Apple Development: stupidmoron@gmail.com (XXXXXXXX)",
  "enable-macos-desktop": false
}

What to do?

Bin the project, empty and cry???

And what a performance system when you build in 480sec maybe don't brag about it...

**Comment by SwiftyFlow** [comment]
Xcode build done.                                           310.6s

**Comment by SwiftyFlow** [comment]
Xcode build done.                                           480.8s

**Comment by SwiftyFlow** [comment]
Nah cos on top of not working it's slow as

**Comment by SwiftyFlow** [comment]
verbose way to long to be pasted I can't even scroll back to the top

podfile:
# Uncomment this line to define a global platform for your project
platform :ios, '12.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end



doctor v:

[✓] Flutter (Channel master, 2.11.0-0.0.pre.452, on macOS 12.1 21C52 darwin-x64, locale en-AU)
    • Flutter version 2.11.0-0.0.pre.452 at /Users/flowpottevin/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ba01ec8faa (2 days ago), 2022-02-05 22:05:18 -0500
    • Engine revision 0712096899
    • Dart version 2.17.0 (build 2.17.0-85.0.dev)
    • DevTools version 2.10.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/flowpottevin/Library/Android/sdk
    • Platform android-31, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.64.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • iPhone 8 (mobile)  • 8f24ab2457ddfe33c15e3d57bf3b3654ac0f9d41 • ios            • iOS 15.1 19B74
    • iPhone 12 (mobile) • 37E4BB48-7DED-4B0F-9DD1-674325FD9C73     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • Chrome (web)       • chrome                                   • web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Comment by darshankawar** [comment]
@SwiftyFlow 
You seem to have shared lot of details but I can't still figure out from it, as what exactly is the issue you are reporting.

Also, please see the code of conduct: https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md#code-of-conduct when you report an issue or comment about it.

Coming back to your issue, in order to properly address it, please elaborate on what exactly the issue you are facing so that we can analyze it and help you on it.

**Comment by SwiftyFlow** [comment]
I went to eat, big mistake now the cursed phone is unusable again. I was all morning... Why is it picking a wrong iOS signing certificate?

**Comment by darshankawar** [comment]
@SwiftyFlow 
From what I understand, the tool seems to be picking an old or wrong signing certificate. If so, check [this](https://github.com/flutter/flutter/issues/43444) similar issue and see if it similar to your case.

**Comment by SwiftyFlow** [comment]
Yes it is "the tool picking an old or wrong signing certificate". The forwarded similar issue is still open and it's been since 2019. The final fix purposed in this thread has been tried dozen of times already. Where as the other tricks: again 'flutter config --clear-ios-signing-cert' does clear anything at all, this 'To remove a setting, configure it to an empty string' when I do It re-picks the wrong one after and the direct Xcode signing team trick does not work.

**Comment by SwiftyFlow** [comment]
Still searching where is this file vim ~/.flutter_settings
To finally put:
{
"ios-signing-cert": "Apple Development: [stupidmoron@gmail.com](mailto:xxx@gmail.com) (XXXXXXXX)",
"enable-macos-desktop": false
}

**Comment by darshankawar** [comment]
Please follow-up in the original issue for further updates. It has P4 priority, meaning a bug and will be fixed per Flutter's bug fix policy.
Closing this as a duplicate.

**Comment by jmagman** [comment]
@SwiftyFlow can you update https://github.com/flutter/flutter/issues/43444 with your findings? That one stalled because I saw the issue once and then never saw it again, I couldn't reproduce it.

**Comment by SwiftyFlow** [comment]
Sadly I cannot tell what happened... I simply gave up on using the cursed device. Worked a bit more on my project, therefore I needed a few more pub.dev packages and after adding some the cursed device appeared to work again. If I remember right I also made sure that the pubspec.yaml has the dependencies cleared of comment and all grouped together 


...
environment:
  sdk: ">=2.15.1 <3.0.0"

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter
  intl: ^0.17.0
  transparent_image: ^2.0.0
  cached_network_image: ^3.2.0
  shared_preferences: ^2.0.12
  firebase_core : ^1.12.0
  cupertino_icons: ^1.0.4
  cloud_firestore: ^3.1.7
  firebase_auth: ^3.3.6
  image_picker: ^0.8.4+5
  firebase_storage: ^10.2.6
  firebase_messaging: ^11.2.6
  sqflite: ^2.0.2
  package_info_plus: ^1.3.0
  flutter_email_sender: ^5.0.2
  csv: ^5.0.1
  provider: ^6.0.2
  path_provider: ^2.0.8
  shake: ^2.0.0
  material_segmented_control: ^3.1.2

dev_dependencies:
  flutter_test:
    sdk: flutter
 
  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^1.0.0
  flutter_launcher_icons: ^0.9.2
...

**Comment by justjcool** [comment]
i am facing the same issue when running app using flutter run command its picking the other certificate instead of which i applied in the Xcode.

**Comment by jmagman** [comment]
@justjcool then can _you_ update https://github.com/flutter/flutter/issues/43444 with your findings? That one stalled because I saw the issue once and then never saw it again, I couldn't reproduce it.

**Comment by FlorentPottevin** [comment]
@justjcool (couldnt signin as swiftyflow) the only thing I can advise you is to forget the device you want to push to. Keep working on your project via simulator (I know a pain if camera or push testing) but you’ll  end up adding one more dependecy in pubspec and it will all magically work again. In parallel double check your indentations in pubspec and make sure you group the dependencies (ie no green comment in between)

**Comment by jmagman** [comment]
...but before you apply any workarounds please give us the requested output in https://github.com/flutter/flutter/issues/43444#issuecomment-1039410544 so we can fix the issue for everyone without needing workarounds.

**Comment by FlorentPottevin** [comment]
I wish I knew more

**Comment by SwiftyFlow** [comment]
Got it just now! It's definitely due to this crap pubspec.yaml file. I still haven't figured exactly what is causing it but just now I had declared my assets as:

assets:
    - lib/assets/


To give me all from 'assets' folder yet the app wasn't finding what's inside the 'vehicles' subfolder so I tried to add it in there:

assets:
    - lib/assets/
    - lib/assets/vehicles


But then I can't run my app on device because it appear that I still have the example signing (which is not true)

Remove - lib/assets/vehicles back to normal

If anyone knows what's up? Indentation? Repetition? it was 12:18? and it's cloudy...?

**Comment by jmagman** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/43444, please make all updates there.  Particularly, we need the output requested in https://github.com/flutter/flutter/issues/43444#issuecomment-1039410544 to make progress.  Locking.
